ko-search/movies-evan-likes/src/App/components/MovieList/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import R from 'ramda';
+import { centeredContainer } from 'sharedStyles.css';
+import { Filters, Results } from '../index';
+import movieData from '../../../movieData.json';
+import MovieItem from './MovieItem';
+import {
+  cacheRequest,
+  createMoviesData,
+  filterByDecade,
+  filterByTitle,
+  getMovieData,
+} from '../helpers/index';
+import { container } from './styles.css';
+
+export default class MovieList extends Component {
+  state = {
+    movies: createMoviesData(movieData),
+  };
+  componentWillMount() {
+    return this.state.movies.forEach(movie => {
+      if (!localStorage.getItem(`${movie.title} ${movie.year}`)) {
+        getMovieData(this.props.url, movie.title)
+          .then(res => res.data)
+          .then(data => {
+            this.setState({
+              ...this.state,
+              movies: this.state.movies.map(
+                movie =>
+                  movie.title === data.Title
+                    ? {
+                      ...movie,
+                      url: data.tomatoURL,
+                      rating: data.imdbRating,
+                    }
+                    : movie,
+              ),
+            });
+            return data;
+          })
+          .then(req => cacheRequest(req))
+          .catch(err => err);
+      } else {
+        const saved = JSON.parse(localStorage.getItem(`${movie.title} ${movie.year}`));
+        this.setState({
+          ...this.state,
+          movies: this.state.movies.map(movie => ({
+            ...movie,
+            url: saved.tomatoURL,
+            rating: saved.imdbRating,
+          })),
+        });
+      }
+    });
+  }
```

Comment:
Generally speaking, I could definately abstract some of the anonymous functions which would at the very least make the code less verbose.

I think normally I would have been using redux, and having that abstracted into a saga would make it much easier to read, and the logic would flow better and read more synchrounously. 

Otherwise I think I would have to re write how I approached this overall and perhaps think of a better architecture, I just wanted to make sure I submitted this to you guys in a reasonable amount of time as I was late because of computer problems.